cmd/kube-apiserver/app/server.go

```diff
@@ -418,6 +419,17 @@ func buildGenericConfig(
 		return
 	}
 
+	if genericConfig.EgressSelector != nil {
+		networkContext := egressselector.Cluster.AsNetworkContext()
+		dialer, egressErr := genericConfig.EgressSelector.Lookup(networkContext)
+		if egressErr != nil {
+			lastErr = egressErr
+			return
+		}
+		// override the contextual dialer to use konnectivity server
+		proxyTransport.DialContext = dialer
```

Comment:
This leaves the tls params with insecureskiptlsverify set from the ssh proxy construction, but egress proxy has proper trust bundles.

Reply:
Hmm I'm pretty sure that the insecureskipverify is for the underlying connection, not for the one with the egress proxy.

Reply:
@deads2k: The insecureskipverify is for the underlying connection, and not for the connection to the proxy. I don't believe this needs to be changed, cc @cheftako, @caesarxuchao for confirmation.

The tls config for egress proxy are defined in the `egress_selector.go` file (https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go#L123)

Reply:
> @deads2k: The insecureskipverify is for the underlying connection, and not for the connection to the proxy. I don't believe this needs to be changed, cc @cheftako, @caesarxuchao for confirmation.

If that's the case I'd like to see a test demonstrating that the ca bundle is honored even with insecureskiptlsverify set. If this is cleanly extracted, I could handle a unit test.  If it's not, then an integration test ought to work.  I'm afraid it's simply not obvious enough to me via inspection.

Reply:
@deads2k: Thanks for your reply. Before discussing any testing, we'd like to learn more about the origins of the insecureskipverify. The proxyTransport affects proxy, aggregator, and admission webhooks. Both [aggregator](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go#L263) and [admission webhooks](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/util/webhook/authentication.go#L60) only extract the DialContext and TLSClientConfig is not touched . [Node/Pod](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/pod/storage/storage.go) proxy directly use this proxyTransport. Does this imply that we don't have TLS when proxying requests? There was this line in the comments

```
Proxying to pods and services is IP-based... don't expect to be able to verify the hostname
```

but we should still be able to do TLS even if certificates are IP based?

Writing a test to verify the ca bundle will be challenging since `egressselector` doesn't expose the information and we'll need spin up a new cluster to have the proxy enabled and running...this will likely only be possible with a e2e test. I'd much rather see if we can get rid of/move the insecureskipverify flag elsewhere.

Reply:
@deads2k: As part of simplifying the amount of certificates needing to be generated, we have decided to use unix domain sockets for communication between apiserver and network proxy since they are guaranteed to be on the same machine. This skips the need for CA bundles completely, and this `insecureskipverify` flag should be irrelevant.